File: temp.py
#!/usr/bin/env python
import json
import requests

# supress Unverified HTTPS request, only do this in a verified environment
import urllib3
urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)

# Address of the WTI device
URI             = "https://"
SITE_NAME       = "rest.wti.com"

# put in the username and password to your WTI device here
BASE_PATH = "/api/v2/status/temperature"
USERNAME  = "rest"
PASSWORD  = "restfulpassword"
AUTH = (USERNAME, PASSWORD)
HEADER = ""

# or if using user tokens put in the User's Token to your WTI device here
#BASE_PATH = "/api/v2/token/status/temperature"
#HEADER = {"X-WTI-API-KEY":"!m+-w-~qo0aq78n=wgyz2c54c365rknj3rnguew8!4mztzx-6j2wlwoonbh4s1cj"}
#AUTH = ""

try:
	r = requests.get(URI+SITE_NAME+BASE_PATH, verify=False, auth=AUTH, headers=HEADER)

	if (r.status_code == 200):
		parsed_json = r.json()

#		Uncomment to see the JSON return by the unit
#		print (parsed_json)

		cszTemperature = parsed_json['temperature']
		cszTemperatureFormat = parsed_json['format']
		cszTimeStamp = parsed_json['timestamp']

		print("Current Temperature is")
		print("----------------------")
		print(cszTemperature+ " " +cszTemperatureFormat + " @ "+cszTimeStamp+"\r\n")
		print(URI+SITE_NAME+BASE_PATH)
	else:
		print(r.status_code)

except requests.exceptions.RequestException as e:
	print ("Exception:")
	print (e)

File: temp.lua
-- Ensure you have luasec for HTTPS support.
-- Install lua-cjson or download json.lua.
-- luarocks install luasec
-- luarocks install lua-cjson

local https = require("ssl.https")
local ltn12 = require("ltn12")
local cjson = require("cjson")  -- You can also use 'json.lua' if preferred
local mime = require 'mime'


-- Address of the WTI device
URI       = "https://"
SITE_NAME = "rest.wti.com"

-- put in the username and password to your WTI device here
BASE_PATH = "/api/v2/status/temperature"
-- BASE_PATH = "/api/v2/token/status/temperature" -- (Used when Token access is required)
-- Define the URL for the API request
local url = URI .. SITE_NAME .. BASE_PATH

local USERNAME = "rest"
local PASSWORD = "restfulpassword"
local credentials = mime.b64(USERNAME .. ":" .. PASSWORD)

-- Set up the headers (if needed)
local headers = {
--    ["X-WTI-API-KEY"] = "usertokenfromWTIdevice", -- (Used when Token access is required)
    ["Authorization"] = "Basic " .. credentials,
    ["Content-Type"] = "application/json"
}

-- Response table to capture the response body
local response_body = {}

-- Make the HTTPS request
print("HTTP request: ", url)
local res, code, response_headers, status = https.request{
    url = url,
    method = "GET",  -- Assuming a GET request based on the endpoint documentation
    headers = headers,
    sink = ltn12.sink.table(response_body)
}

-- Check if the request was successful
if code == 200 then
    -- Combine the response body parts
    local response_str = table.concat(response_body)
    -- Parse the JSON response
    local response_json = cjson.decode(response_str)
    -- print("Parsed JSON Response: ", response_json)
    -- Access specific parts of the response if needed
    -- For example, if the JSON response has a 'temperature' field:
    print("Unit Time:   ", response_json.timestamp)
    print("Temperature: ", response_json.temperature, response_json.format)
else
    print("HTTP request failed with code: ", code)
end
